Let-and implementation

jchen-cs opened the pull request:
Add support for equations of the form `let x=e1 and y=e2...` verification condition generation. Also includes some test cases to test its operation.

SomeInterestingUserName committed:
Generate VCs for let-and

SomeInterestingUserName committed:
Implement and test let-and verification

Adds verification for parallel equations joined with "and", and includes
some simple examples to test.

SomeInterestingUserName committed:
Delete unnecessary printouts and compiled files

jlvargasme commented on examples/test_parallel/parallel_simple.zls:
```diff
@@ -0,0 +1,4 @@
+let node exec () = 
```
Add brief description of what is being tested, it would be good to have the generated vc as a comment.

jlvargasme commented on examples/test_parallel/parallel_simple2.zls:
```diff
@@ -0,0 +1,4 @@
+let node exec () = 
```
Add brief description of what is being tested, it would be good to have the generated vc as a comment.

jlvargasme reviewed (changes requested):
Implementation is good, just added a few comments to adhere to the pull requests CL guidelines

jchen-cs committed:
Add descriptive comments to let-and tests

Add comments describing expected result and generated VCs for the two test files

jchen-cs committed:
Add new let-and test case that fails

parallel_simple3.zls intentionally fails

jchen-cs commented:
Changes have been implemented, awaiting further review
